crates/bevy_reflect/derive/src/reflect_value.rs

```diff
@@ -25,7 +25,7 @@ use syn::{parenthesized, Attribute, Generics, Path};
 /// (in my_crate::bar) Bar(TraitA, TraitB)
 /// ```
 pub(crate) struct ReflectValueDef {
-    #[allow(dead_code)]
+    #[expect(dead_code, reason = "TODO(MrGVSV): Why should this be kept?")]
```

Comment:
I think this field is also unneeded now.

Reply:
I just found out that this field is used when the `documentation` feature is enabled, so I'm going to keep it.

Reply:
Oh good to know! This is exactly why this PR is going to be really useful haha